src/preamble.js

```diff
@@ -1346,7 +1346,11 @@ FHEAP = new Float64Array(TOTAL_MEMORY);
 #endif
 #endif
 #if USE_TYPED_ARRAYS == 2
+#if POINTER_MASKING
+var buffer = new ArrayBuffer(TOTAL_MEMORY + 0x01000000);
```

Comment:
Actually, I think this should work with `ALLOW_MEMORY_GROTH` as well. Just turn the mask into a `var` and also change that in `changeHeap`. May need some tweaking to the proposal, but that since it's not part of the spec currently, that shouldn't be an issue.